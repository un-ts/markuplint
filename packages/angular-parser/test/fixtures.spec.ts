import path from 'node:path'

import { glob } from 'glob'
import { mlTestFile } from 'markuplint'

test('fixtures', async () => {
  const files = await glob(path.resolve(__dirname, 'fixtures/*.html'))
  for (const filePath of files) {
    const { violations } = await mlTestFile(filePath)
    expect(violations).toMatchSnapshot(path.basename(filePath))
  }
})
