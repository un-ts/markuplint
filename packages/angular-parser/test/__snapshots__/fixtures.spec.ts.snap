// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fixtures > attr-duplication.html 1`] = `
[
  {
    "col": 2,
    "line": 3,
    "message": "The attribute name is duplicated",
    "raw": "[type]",
    "ruleId": "attr-duplication",
    "severity": "error",
  },
  {
    "col": 2,
    "line": 4,
    "message": "The attribute name is duplicated",
    "raw": "[attr.type]",
    "ruleId": "attr-duplication",
    "severity": "error",
  },
]
`;

exports[`fixtures > case-sensitive-tag-name.html 1`] = `[]`;

exports[`fixtures > deprecated-element.html 1`] = `
[
  {
    "col": 1,
    "line": 1,
    "message": "The "font" element is obsolete",
    "raw": "<font color="red">",
    "ruleId": "deprecated-element",
    "severity": "error",
  },
]
`;
